tests/unit/bundle_analysis/test_bundle_analysis.py

```diff
@@ -96,6 +96,45 @@ def test_create_bundle_report():
         report.cleanup()
 
 
+def test_bundle_report_asset_filtering():
+    try:
+        report = BundleAnalysisReport()
+        session_id = report.ingest(sample_bundle_stats_path)
+        assert session_id == 1
+
+        assert report.metadata() == {
+            MetadataKey.SCHEMA_VERSION: 1,
+        }
+
+        bundle_reports = list(report.bundle_reports())
+        assert len(bundle_reports) == 1
+
+        bundle_report = report.bundle_report("invalid")
+        assert bundle_report is None
+        bundle_report = report.bundle_report("sample")
+
+        asset_reports = list(bundle_report.asset_reports(
```

Comment:
[Optional but encouraged]: could we fetch the non-filtered assets, then the filtered ones (like you do here) to see this filtering the list? Even a "len = 4" or something suffices